### PR TITLE
[tensorboard] Add strings to image boxes

### PR DESCRIPTION
**Pull request by lanpa**
Addresses https://github.com/pytorch/pytorch/issues/27300

sample usage:
```python
import torch
from torch.utils.tensorboard import SummaryWriter
with SummaryWriter() as w:
     w.add_image_with_boxes('imagebox_label', torch.ones(3, 240, 240) * 0.5,
             torch.Tensor([[10, 10, 100, 100], [101, 101, 200, 200]]),
             global_step=0, labels=['label1', 'label2'])
```
image

cc @sanekmelnikov @orionr

### CODE REVIEW

File: torch/utils/tensorboard/writer.py [hunk 1]
```diff
@@ -583,19 +583,21 @@ def add_images(self, tag, img_tensor, global_step=None, walltime=None, dataforma
...
             img_tensor (torch.Tensor, numpy.array, or string/blobname): Image data
             box_tensor (torch.Tensor, numpy.array, or string/blobname): Box data (for detected objects)
+              box should be represented as [x1, y1, x2, y2].
             global_step (int): Global step value to record
             walltime (float): Optional override default walltime (time.time())
               seconds after epoch of event
             rescale (float): Optional scale override
             dataformats (string): Image data format specification of the form
               NCHW, NHWC, CHW, HWC, HW, WH, etc.
+            labels (list of string): The strings to be show on each bounding box.
```
Comment (orionr) [inline torch/utils/tensorboard/writer.py:600]: Nit: Wording should be `The label to be shown for each bounding box.` (shown with an n and some tweaks)

Comment (lanpa) [reply torch/utils/tensorboard/writer.py]: fixed!

**Review by orionr — APPROVED** [review]

**Comment by orionr** [comment]
Please fix all failing tests but otherwise LGTM!

**Comment by lanpa** [comment]
rebased and got green light. @orionr
